src/routes/PetsPage/PetsPage.js

```diff
@@ -1,54 +1,90 @@
 import React, { Component } from 'react';
-import PetCard from '../../components/PetCard/PetCard';
+import PetList from '../../components/PetList';
 import { Section } from '../../components/Utils/Utils';
 import PetsService from '../../services/petsService';
 
 export default class PetsPage extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      pets: [],
-      petCount: -1  // this is set to prevent the no pets alert before data is loaded
+      pets: null,
+      typeOfAnimal: [],
+      breed: [],
+      sex: '',
+      minAge: '',
+      maxAge: '',
+      more: []
     };
+    this.inputChanged = this.inputChanged.bind(this);
   }
 
   componentDidMount() {
-    PetsService.getPets()
-      .then(pets => this.setState({ pets, petCount: pets.length }))
+    PetsService.getPets({})
+      .then(pets => this.setState({ pets }))
       .catch(error => console.log(error));
   }
 
-  renderPetRow(rowStart) {
-    return (
-      <div key={rowStart} className='row'>
-        {this.state.pets.slice(rowStart, rowStart + 3).map(pet => this.renderPet(pet))}
-      </div>
-    );
+  inputChanged(field, content) {
+    this.setState({ [field]: content }, () => this.handleSubmit());
   }
 
-  renderPet(pet) {
-    return (
-      <PetCard key={pet.PetID} pet={pet} />
-    );
+  handleSubmit = () => {
+    PetsService.getPets(this.getFilters())
+      .then(pets => {
+        this.setState({ pets });
+      })
+      .catch(res => {
+        this.setState({ error: res.error });
+      });
+  }
+
+  getFilters() {
+    const {
+      typeOfAnimal,
+      breed,
+      sex,
+      minAge,
+      maxAge,
+      more
+    } = this.state;
+    const filters = { availability: 'Available' };
```

Comment:
I think for users they only need to see pets that are currently available. For shelter admins I'm planning to add an Availability filter. I'll address that when I get to work on the edit pets feature. Let me know if you think otherwise!